### PR TITLE
[NOD-1406] remove mempool utxo diff

### PR DESCRIPTION
**Pull request by someone235**

### CODE REVIEW

**Comment by someone235** [comment]
Rebuild after update in master

File: domain/mempool/mempool.go [hunk 1]
```diff
@@ -405,221 +404,128 @@ func (mp *TxPool) HaveTransaction(txID *daghash.TxID) bool {
...
-// processRemovedTransactionDependencies processes the dependencies of a
-// transaction tx that was just now removed from the mempool
-func (mp *TxPool) processRemovedTransactionDependencies(tx *util.Tx) {
+func (mp *TxPool) updateBlockTransactionChainedTransactions(tx *util.Tx) {
 	prevOut := appmessage.Outpoint{TxID: *tx.ID()}
 	for txOutIdx := range tx.MsgTx().TxOut {
 		// Skip to the next available output if there are none.
 		prevOut.Index = uint32(txOutIdx)
-		depends, exists := mp.dependsByPrev[prevOut]
+		txD, exists := mp.chainedTransactionByPreviousOutpoint[prevOut]
```
Comment (stasatdaglabs) [inline domain/mempool/mempool.go:489]: Please call this dependency or something other than txD.

File: domain/mempool/mempool.go [hunk 2]
```diff
@@ -405,221 +404,128 @@ func (mp *TxPool) HaveTransaction(txID *daghash.TxID) bool {
...
+	for _, txIn := range tx.MsgTx().TxIn {
+		delete(mp.chainedTransactionByPreviousOutpoint, txIn.PreviousOutpoint)
+	}
 }
 
-// RemoveDoubleSpends removes all transactions which spend outputs spent by the
+// removeDoubleSpends removes all transactions which spend outputs spent by the
 // passed transaction from the memory pool. Removing those transactions then
 // leads to removing all transactions which rely on them, recursively. This is
 // necessary when a block is connected to the DAG because the block may
```
Comment (stasatdaglabs) [inline domain/mempool/mempool.go:521]: Because the block may what?

File: domain/mempool/mempool.go [hunk 3]
```diff
@@ -701,7 +597,7 @@ func (mp *TxPool) checkPoolDoubleSpend(tx *util.Tx) error {
 func (mp *TxPool) CheckSpend(op appmessage.Outpoint) *util.Tx {
 	mp.mtx.RLock()
 	defer mp.mtx.RUnlock()
-	txR := mp.outpoints[op]
+	txR, _ := mp.mempoolUTXOSet.poolTransactionBySpendingOutpoint(op)
```
Comment (stasatdaglabs) [inline domain/mempool/mempool.go:600]: What's txR? Please rename it to something else.

File: domain/mempool/mempool.go [hunk 4]
```diff
@@ -1019,7 +901,7 @@ func (mp *TxPool) maybeAcceptTransaction(tx *util.Tx, rejectDupOrphans bool) ([]
 	}
 
 	// Add to transaction pool.
-	txD, err := mp.addTransaction(tx, txFee, parentsInPool)
+	txD, err := mp.addTransaction(tx, mass, txFee, parentsInPool)
```
Comment (stasatdaglabs) [inline domain/mempool/mempool.go:904]: Please rename txD.

**Review by stasatdaglabs — CHANGES_REQUESTED** [review]

File: domain/mempool/mempool.go [hunk 5]
```diff
@@ -1145,11 +1015,9 @@ func (mp *TxPool) ProcessOrphans(acceptedTx *util.Tx) []*TxDesc {
 //
 // This function is safe for concurrent access.
 func (mp *TxPool) ProcessTransaction(tx *util.Tx, allowOrphan bool) ([]*TxDesc, error) {
-	log.Tracef("Processing transaction %s", tx.ID())
+	log.Criticalf("Processing transaction %s", tx.ID())
```
Comment (stasatdaglabs) [inline domain/mempool/mempool.go:1018]: This should probably be reverted.

**Review by stasatdaglabs — CHANGES_REQUESTED** [review]

**Review by stasatdaglabs — APPROVED** [review]

**Pull request merged**